test-data/unit/check-redefine.test

```diff
@@ -474,3 +474,12 @@ with A() as x:
     reveal_type(x) # N: Revealed type is "builtins.int"
 with B() as x:
     x = 0 # E: Incompatible types in assignment (expression has type "int", variable has type "str")
+
+
+[case testRedefineModuleAsException]
```

Comment:
Note that if you run this in real life you get:
```
crash.py:5: error: Incompatible types in assignment (expression has type "Exception", variable has type Module)
Found 1 error in 1 file (checked 1 source file)
```
I think the fixtures just aren't up to scratch here.